subprojects/files/src/main/java/org/gradle/api/internal/file/archive/impl/FileZipInput.java

```diff
@@ -61,7 +71,7 @@ private FileZipInput(File file) {
         try {
             this.file = new ZipFile(file);
         } catch (IOException e) {
-            throw new UncheckedIOException(e);
+            throw new FileException(e);
```

Comment:
Any reason for not throwing `java.io.UncheckedIOException`?

Reply:
files is used by worker processes which means it can't use JDK8 classes like `java.io.UncheckedIOException`.

Reply:
Got it, let's do it like this then.